onnxruntime/core/providers/cuda/cuda_allocator.cc

```diff
@@ -36,7 +36,8 @@ void* CUDAAllocator::Alloc(size_t size) {
   CheckDevice(true);
   void* p = nullptr;
   if (size > 0) {
-    CUDA_CALL_THROW(cudaMalloc((void**)&p, size));
+    //Return nullptr when allocation failed. Don't throw exception, the bfc arena will try to adjust the request size.
+    CUDA_CALL(cudaMalloc((void**)&p, size));
   }
```

Comment:
Technically this change isn't needed as BFCArena was updated recently to handle the exception and adjust the request size, so the behavior there will be the same either way. It's a policy choice as to whether this should throw (slightly more consistent with CPU allocator on Windows which throws std::bad_alloc, but not sure about linux).

Reply:
will revert